babel/core.py

```diff
@@ -1193,22 +1240,23 @@ def parse_locale(identifier: str, sep: str = '_') -> tuple[str, str | None, str
     if parts:
         raise ValueError(f"{identifier!r} is not a valid locale identifier")
 
-    return lang, territory, script, variant
+    return lang, territory, script, variant, (modifier or None)
```

Comment:
Actually, can you change this function to only return a 5-tuple if there is a modifier, and then add a `# TODO(3.0): always return a 5-tuple` sort of comment? We're trying pretty hard to not change public APIs.